### ISSUE TITLE
pcloud Can't download file

### ISSUE THREAD

**Issue by MilosDjonovic** [pCloud/pcloud-sdk-js#6]
I've try example for download:
https://github.com/pCloud/pcloud-sdk-js/blob/master/examples/download.html
and it does not work.
When I clikc on "Get token", files are listed, but download button does not work.
Console.log:
Uncaught (in promise) {result: 7010, error: "Invalid link referer."}

**Comment by sorentycho** [comment]
See the other thread: https://github.com/pCloud/pcloud-sdk-js/issues/4#issuecomment-439262472

**Comment by MilosDjonovic** [comment]
I saw that.
I want to check that it is not possible to download the file with js sdk

**Comment by sorentycho** [comment]
From what I can tell looking through the code (and the error you recieved), the JS SDK is using the same HTTP API we've been trying. So I think, even though the SDK includes a `browser` module, it's not possible.

I'm porting my app to Dropbox, which I've just verified to allow file downloading in the browser on an arbitrary site. Dunno if that helps you or not.

**Comment by MilosDjonovic** [comment]
It helps.
Thanks!

**Comment by SirMartin** [comment]
The problem (I had the same after long work time) is that the api creates links to download files/images/etc for a determinated IP, so if you generate the link (as I made) from your server side (the link is limited to the server IP), so if you try to consume the link from the client side (as also I made), it doesn't work, because are not available.

And yes is really disappointed that this restriction is not documented in any place, and also that you cannot disable it. For me worked fine for weeks that I only tried in localhost, and when I was ready to publish it, boom, it doesn't work.

**Comment by igneosaur** [comment]
The issue is that the method `downloadfile` also uses the [`getfilelink` method](https://github.com/pCloud/pcloud-sdk-js/blob/master/src/client/methods/downloadfile.js#L13) which ["can't be used from web applications. Referrer is restricted to pcloud.com"](https://docs.pcloud.com/methods/streaming/getfilelink.html). See #4.

**Comment by ruffin--** [comment]
Yeah, it looks like that download.html file should've been removed. I got the token part working well (see [this PR](https://github.com/pCloud/pcloud-sdk-js/pull/55)), but then ran into your run-of-the-mill CORS errors.

As @igneosaur remarks, `getfilelink` only works for pCloud. 

<img width="495" alt="This method can't be used from web applications. Referrer is restricted to pcloud.com." src="https://user-images.githubusercontent.com/5659878/188544185-f8581677-4458-4c45-82c7-0f65f8c16935.png">

You're going to need a proxy [like what's mentioned here](https://stackoverflow.com/a/56781665/1028230), but one you're hosting yourself, I believe.

**Comment by alensiljak** [comment]
I have a PWA app which gets access to its app folder only and is used to backup and (was hoping) restore the data and settings to/from pCloud.
This is pretty bad news, finding it out only once coming to the point where a download needs to be made from the app. 
Subscribing to the issue in hope things improve (and soon).
